elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/elements/TypedOperation.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.contrib.testhelpers.graphql.elements;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.io.Serializable;
+
+/**
+ * According to GraphQL grammar (6.0):
+ * <pre>
+ * {@code
+ * operationType  name? variableDefinitions? directives? selectionSet;
+ *
+ * operationType : SUBSCRIPTION | MUTATION | QUERY;
+ * }
+ * </pre>
+ * A {@link TypedOperation} comes with required {@code operation type} and {@link SelectionSet} along with optional
+ * operation name and variables used in query.
+ */
+@RequiredArgsConstructor
+public class TypedOperation implements OperationDefinition {
+
+    /**
+     * Models the required {@code operation type}.
+     */
+    @RequiredArgsConstructor
+    public enum OperationType implements Serializable {
```

Comment:
Instead of having an enum here that is a member of a class - I think it would be simpler/cleaner to just have classes for Mutation, Subscription (which we can leave out for now since we don't use it), and Query.  Also, I'm a bit confused by OperationDefinition and Definition.  I realize they are concepts referenced in GraphQL - but I'm not sure we really need them.

Reply:
I can remove `OperationDefinition`, but if we also remove `Definition`, then I think it will become a very loose type system filled with Objects that is going to make DSL entirely loose compile-time check.